fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -251,14 +256,36 @@ private List<RemoteFileInfo> getRemoteFileInfos(IcebergTable table, long snapsho
                 scan = scan.filter(icebergPredicate);
             }
 
-            CloseableIterable<CombinedScanTask> combinedScanTasks = scan.planTasks();
-            for (CombinedScanTask combinedScanTask : combinedScanTasks) {
-                for (FileScanTask fileScanTask : combinedScanTask.files()) {
-                    builder.add(fileScanTask);
+            CloseableIterable<FileScanTask> fileScanTaskIterable = TableScanUtil.splitFiles(
+                    scan.planFiles(), scan.targetSplitSize());
+            CloseableIterator<FileScanTask> fileScanTaskIterator = fileScanTaskIterable.iterator();
+            Iterator<FileScanTask> fileScanTasks;
+
+            // Under the condition of ensuring that the data is correct, we disabled the limit optimization when table has
+            // partition evolution because this may cause data diff.
+            boolean canPruneManifests = limit != -1 && !table.isV2Format() && onlyHasPartitionPredicate(table, predicate)
```

Comment:
If it's V2, we can use this optimization if the file doesn't attach with stats file